Userland/Libraries/LibGfx/ImageFormats/JPEGWriter.cpp

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "JPEGWriter.h"
+#include "JPEGShared.h"
+#include "JPEGWriterTables.h"
+#include <AK/BitStream.h>
+#include <AK/Endian.h>
+#include <AK/Function.h>
+#include <LibGfx/Bitmap.h>
+
+namespace Gfx {
+
+namespace {
+
+// This is basically a BigEndianOutputBitStream, the only difference
+// is that it appends 0x00 after each 0xFF when it writes bits.
+class JPEGBigEndianOutputBitStream : public Stream {
+public:
+    explicit JPEGBigEndianOutputBitStream(MaybeOwned<Stream> stream)
```

Comment:
It's shame that we have to copy over the entire `BigEndianOutputBitStream` class, but I don't see a quick way out of this. Although maybe we could get rid of the `MaybeOwned` here? In the context of `JPEGWriter`, you always have a `Stream&`.

Reply:
> It's shame that we have to copy over the entire BigEndianOutputBitStream class, but I don't see a quick way out of this.

I wasn't happy to do it, but we have the same issue with the encoder and never found a solution either.

> Although maybe we could get rid of the MaybeOwned here?
You're right, let me fix that.